lumeer-core/src/main/java/io/lumeer/core/facade/LinkInstanceFacade.java

```diff
@@ -77,14 +77,17 @@ public void init() {
    }
 
    public LinkInstance createLinkInstance(final LinkInstance linkInstance) {
-      checkLinkTypeWritePermissions(linkInstance.getLinkTypeId());
+      var linkType = checkLinkTypeWritePermissions(linkInstance.getLinkTypeId());
 
       linkInstance.setCreatedBy(authenticatedUser.getCurrentUserId());
       linkInstance.setCreationDate(ZonedDateTime.now());
       LinkInstance createdLinkInstance = linkInstanceDao.createLinkInstance(linkInstance);
 
-      final DataDocument data = linkDataDao.createData(linkInstance.getLinkTypeId(), createdLinkInstance.getId(), new DataDocument());
-      createdLinkInstance.setData(data);
+      var data = linkInstance.getData();
+      constraintManager.encodeDataTypes(linkType, data);
+
+      var storedData = linkDataDao.createData(linkInstance.getLinkTypeId(), createdLinkInstance.getId(), data);
+      createdLinkInstance.setData(storedData);
```

Comment:
You are missing decoding data before returning it. (Look at `DocumentFacade.createDocument` function)